flink-table/flink-sql-parser/src/main/codegen/includes/parserImpls.ftl

```diff
@@ -2296,3 +2296,38 @@ SqlNode SqlAnalyzeTable():
         return new SqlAnalyzeTable(s.end(this), tableName, partitionSpec, columns, allColumns);
     }
 }
+
+/**
+* Parses a STOP JOB statement:
+* STOP JOB <JOB_ID> [<WITH SAVEPOINT>] [<WITH DRAIN>];
+*/
+SqlStopJob SqlStopJob() :
+{
+    SqlCharStringLiteral jobId;
+    boolean isWithSavepoint = false;
+    boolean isWithDrain = false;
+}
+{
+    <STOP> <JOB> <QUOTED_STRING>
+    {
+        String id = SqlParserUtil.parseString(token.image);
+        jobId = SqlLiteral.createCharString(id, getPos());
+    }
+    [
+        LOOKAHEAD(2)
+        <WITH> <SAVEPOINT>
```

Comment:
It seems it's impossible to parse 

```
STOP JOB 'myjob'  WITH DRAIN WITH SAVEPOINT
```

I think we can do as `EXPLAIN` syntax that allows users to specify `ExplainDetail` in any order.

Reply:
I've reconsidered the syntax of `WITH DRAIN`, and found `WITH DRAIN` should be an attribute to `WITH SAVEPOINT`. That's to say, we could not have `WITH DRAIN` without `WITH SAVEPOINT`. 

How about we keep the order and validate that `WITH DRAIN` could only appear after `WITH SAVEPOINT`?